### DISCUSSION TITLE
SMTP email error on signup

### DISCUSSION THREAD

**Discussion by BackedUpBooty** [ente-io/ente#1582] [Q&A]
### Description

Trying to set this up for the first time, using docker, image: ghcr.io/ente-io/server (not build). 

Containers start properly, Android app can connect to my local endpoint and via reverse proxy. Attempting to create a new account returns an error (see screenshot), phone logs show an HTTP status error 500 and the container logs show the following:

```
INFO[0071]request_logger.go:83 func1 incoming                                      client_ip=192.168.1.20 client_pkg=io.ente.photos client_version=0.8.80 query= req_body={"email":"[redacted email]","purpose":""} req_id=b48b2a97-1582-4da7-9723-a0234a4f4cfe req_method=POST req_uri=/users/ott ua=Dalvik/2.1.0 (Linux; U; Android 14; SM-S918B Build/UP1A.231005.007)
INFO[0071]userauth.go:125 SendEmailOTT Added ott for Kv9gU8MBdNOehZ+jq+9D8020/Y/93DVjYLpShjhikG8=: 769048 
ERRO[0075]handler.go:46 Error Request failed                                error= --- at /etc/ente/pkg/api/user.go:40 (UserHandler.SendOTT) ---
 --- at /etc/ente/pkg/controller/user/userauth.go:128 (UserController.SendEmailOTT) ---
 --- at /etc/ente/pkg/controller/user/userauth.go:309 (emailOTT) ---
 --- at /etc/ente/pkg/utils/email/email.go:88 (sendViaSMTP) ---
Caused by: 550 Can't send from verification@ente.io with user [redacted email] req_id=b48b2a97-1582-4da7-9723-a0234a4f4cfe user_id=0
ERRO[0600]handler.go:46 Error Request failed                                error= --- at /etc/ente/pkg/api/user.go:40 (UserHandler.SendOTT) ---
INFO[0597]userauth.go:125 SendEmailOTT Added ott for Kv9gU8MBdNOehZ+jq+9D8020/Y/93DVjYLpShjhikG8=: 310507 
INFO[0597]request_logger.go:83 func1 incoming                                      client_ip=192.168.1.20 client_pkg=io.ente.photos client_version=0.8.80 query= req_body={"email":"ente@pointtosource.com","purpose":""} req_id=63a6bc62-e34c-4740-b4b3-6679c6374f1c req_method=POST req_uri=/users/ott ua=Dalvik/2.1.0 (Linux; U; Android 14; SM-S918B Build/UP1A.231005.007)
Caused by: 550 Can't send from verification@ente.io with user ente@pointtosource.com req_id=63a6bc62-e34c-4740-b4b3-6679c6374f1c user_id=0
```

![image](https://github.com/ente-io/ente/assets/53868636/dcca0e20-dda1-4c59-a315-09ac0e1a4bd6)

credentials.yaml is being used, key.encryption, key.hash and jwt.secret all created with base64 but the error persists even if they aren't in use. SMTP credentials are correct. 

When trying with gmail smtp it shows that it was unauthorized, but again the credentials are correct. Escaping the password makes no difference. 

I've tried setting the variable directly in the compose environment block (`ENTE_SMTP_HOST=`, `ENTE_SMTP_PORT=` etc.) though to be honest I didn't expect that to work either. 

What am I missing? 

### Version

latest

### What product are you using?

Ente Photos

### What platform are you using?

Mobile - Android, Server

**Comment by mnvr** [comment]
Please don't open issues related to self hosting. I've moved this to a discussion.

Now in terms of your actual issue, three things come to mind:

1. `ghcr.io/ente-io/server` lags a bit behind the main server code, so you could build from source to rule that out as an issue. However, AFAIR, there have not been any fixes to the SMTP code since the ghcr.io build (commit `7be2c66fb69517eb1a2f67b0d94cc72817687df9` so I don't think this is the problem here)

2. There's something wrong with the credentials. You should verify using them with some other SMTP client. I'm not sure about Gmail, but many providers will insist that you use an app-specific password instead of your actual email password if trying to connect over a automated connection. Or you might've 2FA turned on. etc.

3. Gmail might not like that you're trying to send from `verification@ente.io` instead of your actual email address. If this is indeed the issue, then you'll need to build from source, there isn't currently a configuration flag to change the source email address.

If you ask me, I think 2 is the issue. If not, then 3. For reference, we ourselves are using the SMTP flow on production, and it's working fine.

---

BTW, if all you need is the OTP to log in, see https://help.ente.io/self-hosting/faq/otp

**Reply by BackedUpBooty** [reply]
Thanks and apologies, I must have skipped an instruction somewhere about where to post my problem. 

I'll check with my smtp forwarder, but the credentials are correct (created more than one email with password). The Google app-specific password is probably the issue, should have thought of that. 

And OTP - that will definitely help, but I'll keep trying to get the email working too. Appreciate the quick reply.

**Reply by Shackelford-Arden** [reply]
@BackedUpBooty I too was a little confused about issues not being preferred. I guess it is to be inferred from this section:

https://help.ente.io/self-hosting/#next-steps

that discussions are intended to be the place for self-hosted items, though maybe it can be more explicitly listed in the docs there that opening actual issues is not the correct path, @mnvr :eyes:?

**Reply by mnvr** [reply]
It's in the [README](https://github.com/ente-io/ente/tree/main/server#self-hosting)

> Some people new to Docker/Go/Postgres might have general questions though. Unfortunately, because of limited engineering bandwidth we will currently not be able to prioritize support queries related to self hosting, and we request you to please not open issues around self hosting for the time being (feel free to create discussions though). The best way to summarize the status of self hosting is – everything you need is here, but it is perhaps not readily documented, or flexible enough.

**Reply by Shackelford-Arden** [reply]
Fair point!

**Reply by pavelzw** [reply]
I also have a similar issue 

```
ERRO[0063]handler.go:47 Error Request failed                                error= --- at /etc/ente/pkg/api/user.go:40 (UserHandler.SendOTT) ---
 --- at /etc/ente/pkg/controller/user/userauth.go:128 (UserController.SendEmailOTT) ---
 --- at /etc/ente/pkg/controller/user/userauth.go:310 (emailOTT) ---
 --- at /etc/ente/pkg/utils/email/email.go:88 (sendViaSMTP) ---
Caused by: 553 5.7.1 <verification@ente.io>: Sender address rejected: not owned by user my@email.com req_id=... user_id=0
```

Most probably because of 3.
It would be nice if at some point it was possible to specify the hardcoded emails like `verification@ente.io`, `team@ente.io`, etc.
https://github.com/ente-io/ente/blob/944a075fb14009d5cd0d5354ad732596456647cd/server/pkg/controller/user/userauth.go#L305

**Reply by mnvr** [reply]
That does seem like because of 3. We've now added an option to provide your own email address to use as the sender address - https://github.com/ente-io/ente/blob/e96c9dd2a30b0e5f7705ccaa8ea88045a58811ee/server/configurations/local.yaml#L185 - when this option is present, it'll always be used instead of those hardcoded emails.

**Comment by janstadt** [comment]
Does anyone have an example of getting smtp configured for gmail? Do the gmail smtp values go into museum.yaml or credentials.yaml? It wont build for me if there in museum but when i have them in credentials i hit this error above.

**Reply by janstadt** [reply]
actually, it builds in museum.yaml. i just mistakenly had some env vars in there and had to hard code the creds instead of passing them in from my `.env` file and compose file.
